src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/ConfigField.java

```diff
@@ -34,27 +41,66 @@
     }
 
     public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group) {
-        this(key, label, type, required, sensitive, group, "");
+        this(key, label, type, required, sensitive, group, "", null);
+    }
+
+    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group, final BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction) {
+        this(key, label, type, required, sensitive, group, "", validationFunction);
     }
 
     public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final String subGroup) {
-        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, subGroup);
+        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, subGroup, null);
+    }
+
+    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final String subGroup, final BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction) {
+        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, subGroup, validationFunction);
     }
 
     public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive) {
-        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, "");
+        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, "", null);
+    }
+
+    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction) {
+        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, "", validationFunction);
+    }
+
+    public Collection<String> validate(final FieldValueModel fieldToValidate, final FieldModel fieldModel) {
+        List<BiFunction<FieldValueModel, FieldModel, Collection<String>>> validationFunctions = List.of();
+        if (null != validationFunction) {
+            validationFunctions = List.of(validationFunction);
+        }
+        return validate(fieldToValidate, fieldModel, validationFunctions);
+    }
+
+    Collection<String> validate(final FieldValueModel fieldToValidate, final FieldModel fieldModel, final List<BiFunction<FieldValueModel, FieldModel, Collection<String>>> validationFunctions) {
+        final Collection<String> errors = new LinkedList<>();
+        final boolean performValidation = !fieldToValidate.isSet() || fieldToValidate.hasValues();
+        if (performValidation) {
+            if (fieldToValidate.hasValues()) {
+                for (final BiFunction<FieldValueModel, FieldModel, Collection<String>> validation : validationFunctions) {
```

Comment:
How does the default validation make it into this? Or do we not want the default validation if someone providers alternative(s)?